lib/uri_checker.rb

```diff
@@ -0,0 +1,213 @@
+require "uri"
+
+module UriChecker
```

Comment:
I think it'd make sense that the link checking classes follow a namespace (or series of them) so I'd guess at LinkCheck::UriChecker.

I think they'd also be better separated out to multiple files as this has become quite a long file and could be confusing to find the classes